lib/reverse_sentence.rb

```diff
@@ -1,4 +1,77 @@
+require 'pry'
 # A method to reverse the words in a sentence, in place.
-def reverse_sentence(my_sentence)
-  raise NotImplementedError
+def reverse_sentence(my_words)
+
+  string_reverse(my_words)
+
+  if my_words == nil
+    return nil
+  end
+
+  length = (my_words.length - 1)
+  i = 0
+  while i < length
+
+    while my_words[i] == " "
+      i = i + 1
+    end
+
+    start = i
+
+    while my_words[i] != " "
+      i = i + 1
+      break if my_words[i].nil?
```

Comment:
Consider using my_words.length instead of checking for the character value being nil. That way, your algorithm is less reliant on the language specific constructs.